common/include/pcl/pcl_base.h

```diff
@@ -54,11 +54,12 @@
 #include <pcl/point_cloud.h>
 #include <pcl/PointIndices.h>
 #include <pcl/PCLPointCloud2.h>
+#include <pcl/types.h>
 
 namespace pcl
 {
   // definitions used everywhere
-  using Indices = std::vector<int>;
+  using Indices = PointIndices::Indices;
```

Comment:
As I said, I think `Indices` should be declared as `std::vector<index_t>`, and then `PointIndices::Indices` aliased to `Indices`. Also `types.h` may be more appropriate place for `Indices`.

Reply:
Yeah. I was on similar thought process.

Reply:
Indices should be a `std::span<index_t>`, not `vector<index_t>`

Reply:
...which is a C++20 feature? Also, come on, another breaking change?

Reply:
So I messed up my thought process. Starting from scratch here:

We need 2 different indice type:
* Owning: for storing indices (This fits the `Indices` name better, there's no breaking change)
* Non-owning: for taking in indices as function parameters (This needs some name... `IndicesView` sounds weird)

This will allow `std::vector<index_t, custom_allocator>` to work without having to add templates to everything.

`span` need not be from `c++20` but can be borrowed from open implementations like gsl or span-lite, etc.

Reply:
OK, maybe you can put this into RFC or something :) Not for this PR.